wordpress-mobile/WordPress-iOS
# Feature/10323 verticals wizard

Implements #10323 in part.

The UI is far from acceptable, but in the interest of solving the UI flow as soon as posible, I am opening this PR for discussion now.

In this PR:
- Vertical step
- A VC to render the content of this step
- Scaffolding for navigation. It is possible to trigger navigation from one WizardStep, to another WizardStep, by passing its delegate the identifier of the destination step
- Some abstractions are starting to popup. For example, a generic datasource that we can use to render both the tableview in the Segments and Verticals steps.
- Add properties to the SiteCreator accordingly.

My main takeaway from this PR is that the SegmentsStep and VerticalsStep are not that different. And they are not that different from the AddressStep. I think we should be able to solve the three of them by a combination of generics and extracting the behaviour that is actually different, which is not that much. So I expect changes in the next couple of PRs, particularly with the implementation of the AddressStep, which is what I am going to attack now.

To test:
- Run the tests
- Navigate to My Sites > tap + to add a site > Enhanced site creation
- Select any Segment, it should navigate to the Vertical screen. 
- Type anything in the textfield, the table view should be populated with mock data
